Server/src/controllers/VariablesController.ts

```diff
@@ -0,0 +1,33 @@
+import { Response } from 'express';
+import { BadRequest } from '@tsed/exceptions';
+import { VariableService } from '../services/VariableService';
+
+/**
+ * This controller is responsible for requests related to the variables route
+ */
+export class VariablesController {
+  private variableService: VariableService;
+  private invalidResponse: boolean;
+
+  constructor() {
+    this.variableService = new VariableService();
+  }
+
+  async retrieveVariables(response: Response): Promise<Response> {
+    try {
+      let requestResponse = await this.variableService.processGetAllVariables();
+      return response.status(requestResponse.statusCode).json(requestResponse.message)
+    } catch (error) {
+      this.handleError(response, error);
+    }
+  }
+
+  private handleError(response: Response, error: any) {
```

Comment:
Much nicer. We should do this with most of our error handling

Reply:
:+1: